tests/test_pi_extension_dispatch.py

```
@@ -0,0 +1,383 @@
+
+
+def make_executable(path: Path, content: str) -> None:
+    path.write_text(content, encoding="utf-8")
+    path.chmod(0o755)
+
+
+def install_extension(root: Path) -> Path:
+    config_dir = root / "pi-agent"
+    env = os.environ.copy()
+    env["PI_CODING_AGENT_DIR"] = str(config_dir)
+    install = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)